server/src/main/java/org/opensearch/cluster/metadata/MetadataUpdateSettingsService.java

```diff
@@ -137,15 +138,17 @@ public void updateSettings(
         for (String key : normalizedSettings.keySet()) {
             Setting setting = indexScopedSettings.get(key);
             boolean isWildcard = setting == null && Regex.isSimpleMatchPattern(key);
+            boolean isArchived = key.startsWith(ARCHIVED_SETTINGS_PREFIX);
             assert setting != null // we already validated the normalized settings
+                || isArchived
                 || (isWildcard && normalizedSettings.hasValue(key) == false) : "unknown setting: "
                     + key
                     + " isWildcard: "
                     + isWildcard
                     + " hasValue: "
                     + normalizedSettings.hasValue(key);
             settingsForClosedIndices.copy(key, normalizedSettings);
-            if (isWildcard || setting.isDynamic()) {
+            if (isWildcard || isArchived || setting.isDynamic()) {
```

Comment:
we shouldn't expose removing archive settings for open indices. It can cause change in behavior. It would be reasonable to close the index, remove the setting and reopen it again.

Reply:
if a setting is archived, it inherently implies that the cluster doesn't recognise this setting and shouldn't have any impact. Opening and closing the index for these unrecognized settings might be more disruptive(availability wise). Let me know what you think.

Reply:
If there is an archived setting, it means active cluster manager is not able to recognize. If the plugin is available to data nodes, it can potentially recognize it. Also there could be a case post rolling restart, plugin got removed from the node while the index was open.

Reply:
If the cluster manager doesn't recognize the setting, setting would be moved under `archived` prefix. 
So if a node contains the plugin which does recognize the setting, it would not find the setting in IndexMetadata(cluster state) and resort to the default value. Removal of archived setting would have no impact here. 

Hypothetically, if the setting was static and already set on the index, and then master moves it to archived, node containing the shard could still be  operating with the original value(need to verify this). 
Technically, in this case also, removing the setting doesn't disrupt. The shard would disrupt when it gets closed and reopened. I guess we can add the restriction as a safety measure and remove the archived settings only on closed indices.

Reply:
Like discussed offline, added the restriction to only modify archived index settings on closed indices.